### PR TITLE
fetch branch for bitbucket onprem

### PR DESCRIPTION
**Pull request by devkimittal**

### CODE REVIEW

File: scm/driver/stash/testdata/default_branch.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+    "id": "refs/heads/mohit/mohit-temp",
```
Comment (rutvijmehta-harness) [inline scm/driver/stash/testdata/default_branch.json:2]: nit: Prefer adding generic names

File: scm/driver/stash/testdata/repo.json.golden [hunk 2]
```diff
@@ -3,7 +3,7 @@
     "Namespace": "PRJ",
     "Name": "my-repo",
     "Perm": null,
-    "Branch": "master",
+    "Branch": "notmaster",
```
Comment (rutvijmehta-harness) [inline scm/driver/stash/testdata/repo.json.golden:6]: nit: feature_branch

File: scm/driver/stash/testdata/content_list.json [hunk 3]
```diff
@@ -22,6 +22,7 @@
     "stash.go",
     "stash_test.go",
     "testdata/branch.json",
+    "testdata/default_branch.json",
```
Comment (rutvijmehta-harness) [inline scm/driver/stash/testdata/content_list.json:25]: Don't you have to add golden as well here?

nit: Alphabetical order

File: scm/driver/stash/testdata/content_list.json.golden [hunk 4]
```diff
@@ -75,6 +75,10 @@
     "path": "testdata/branch.json",
     "kind": "file"
   },
+  {
```
Comment (rutvijmehta-harness) [inline scm/driver/stash/testdata/content_list.json.golden:78]: ditto

**Comment by devkimittal** [comment]
@rutvijmehta-harness made the changes according to your comments.

**Review by rutvijmehta-harness — APPROVED** [review]

**Pull request merged**